### PR TITLE
chore(docs): add note about Helm Chart

### LINKED ISSUE #27476: Volume is missing for sshKeySecret when dag persistence is enabled.
### Official Helm Chart version

1.7.0 (latest released)

### Apache Airflow version

2.3.4

### Kubernetes Version

1.22.6

### Helm Chart configuration

```
# Git sync
dags:
  persistence:
    enabled: true
    size: 1Gi
    storageClassName:
    accessMode: ReadWriteOnce
    existingClaim: airflow-dags
    subPath: ~
  gitSync:
    enabled: true
    repo: "ssh://git@..."
    branch: "main"
    rev: HEAD
    depth: 1
    maxFailures: 0
    subPath: ""
    sshKeySecret: airflow-ssh-secret
```

### Docker Image customisations

_No response_

### What happened

This commit: https://github.com/apache/airflow/commit/3fc895b9dfe8e7b77538bd80754fb17ccf92db49 causes the following error because volumeMount is created but the volume is missing:
```upgrade.go:369: [debug] warning: Upgrade "airflow" failed: cannot patch "airflow-scheduler" with kind Deployment: Deployment.apps "airflow-scheduler" is invalid: spec.template.spec.containers[2].volumeMounts[0].name: Not found: "dags"```

### What you think should happen instead

I don't see any relation between dag persistency and git-sync configuration. The fact is that the same configuration works when `credentialsSecret` is defined but stopped working with `sshKeySecret`.

### How to reproduce

Follow commit https://github.com/apache/airflow/commit/3fc895b9dfe8e7b77538bd80754fb17ccf92db49

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by aairey**
The Helm Chart does not support both git-sync and persistence to be configured at the same time.

Fixes #27476

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: docs/helm-chart/manage-dags-files.rst [hunk 1]
```diff
@@ -144,6 +144,11 @@ hence Webserver does not need access to DAG files, so ``git-sync`` sidecar is no
 Notes for combining git-sync and persistence
 ............................................
 
+.. note::
+
+    When using the helm chart, it is not possible to use git-sync and persistence at the same time
```
Comment (amoghrajesh) [inline docs/helm-chart/manage-dags-files.rst:149]:
```suggestion
    When using the helm chart, it is not recommended to use git-sync and persistence at the same time
```

**Review by amoghrajesh — COMMENTED** [review]
Keeping the note aligned with the below reasoning

File: docs/helm-chart/manage-dags-files.rst (hunk 1 above)
Comment (aairey) [reply docs/helm-chart/manage-dags-files.rst:149]: I believe @mashanm mentioned it is technically not even possible to use both at the same time when using the Helm Chart.

And this is undocumented.

Comment (romsharon98) [reply docs/helm-chart/manage-dags-files.rst:149]: if it's not possible we should change this line too.
`While using both git-sync and persistence for DAGs is possible`

**Comment by aairey** [comment]
> Keeping the note aligned with the below reasoning

Sorry it is a bit unclear to me what is expected now.
Could you suggest your changes inline?

**Comment by shalberd** [comment]
@potiuk The content of this PR is completely valid, yet it was never approved and merged. Weird, to say the least.
On the one hand, you demand PRs in #27476, on the other hand, when the user does submit one, it is not taken seriously ...

File: docs/helm-chart/manage-dags-files.rst (hunk 1 above)
Comment (shalberd) [reply docs/helm-chart/manage-dags-files.rst:149]: @aairey changed it accordingly, yet the PR was never merged ...

**Pull request closed without merging**